app/controllers/users/backup_code_setup_controller.rb

```diff
@@ -11,12 +11,12 @@ class BackupCodeSetupController < ApplicationController
     before_action :set_backup_code_setup_presenter
     before_action :apply_secure_headers_override
     before_action :authorize_backup_code_disable, only: [:delete]
-    before_action :confirm_recently_authenticated_2fa, except: [:reminder]
 
     helper_method :in_multi_mfa_selection_flow?
 
     def index
```

Comment:
I don't think we want any changes in this file for this pull request. Rather than updating `#index` to render `#create`, we should update anything which redirects the user to `#index` to redirect to `#create` instead.

The idea with doing this in two pull requests is to stop all of the traffic being sent to `GET /backup_code_setup` so that we can safely remove the route in the follow-on pull request. But the changes here currently do not stop traffic being sent to `/backup_code_setup`.

Reply:
@aduth I tried updating the confirmation_path here: https://github.com/18F/identity-idp/blob/main/app/controllers/concerns/mfa_setup_concern.rb#L17-L34 by changing the 'backup_code' case to backup_code_create_path which seemed appropriate given how it's represented in the routes (/rails/info/routes) as users/backup_code_setup#create. 
My suspicion for why is it isn't working is and goes to #index regardless because confirmation_path here: https://github.com/18F/identity-idp/blob/main/app/controllers/users/two_factor_authentication_setup_controller.rb#L68-L70 using 'redirect_to' sends a GET request but in order to get to '#create' it needs to be PATCH. Does that sound right or can/should I somehow add a status code that would make the desired http request?
https://dzone.com/articles/rails-redirect-to

Reply:
Hm, that's a good point about how we'll now be redirecting a user immediately to the mutative `PATCH #create` route via `GET`. Generally I think this makes more sense as a `PATCH #create` since it's mutating / creating something (the codes), but given the difficulty in the redirection, maybe we can go at it from the opposite approach: Have `GET #index` handle the creation of the backup codes (i.e. everything that `#create` is doing today), and then phase out / remove the `PATCH #create` route. What do you think?

We could also do something where `GET #index` auto-submits or otherwise redirects to `PATCH #create`, but I'm not sure how well that would work, especially if "auto-submit" implies some JavaScript requirement (wouldn't work if JavaScript was disabled).

Reply:
> Have GET #index handle the creation of the backup codes (i.e. everything that #create is doing today), and then phase out / remove the PATCH #create route. What do you think?

That sounds like it would work. I'd have to do that in the backup_code_setup_controller, correct?
I think the JS way like you said breaks on no-js but also maybe adds some overhead that could be avoided by just letting the controller handle the job.

Reply:
> I'd have to do that in the backup_code_setup_controller, correct?

Yeah, and if it's a short-term thing, maybe we just copy-paste the contents of [`def create`](https://github.com/18F/identity-idp/blob/b54b3c8a385f1acf8f74cc3565b107026f7c69ea/app/controllers/users/backup_code_setup_controller.rb#L22-L31) to replace [`def index`](https://github.com/18F/identity-idp/blob/b54b3c8a385f1acf8f74cc3565b107026f7c69ea/app/controllers/users/backup_code_setup_controller.rb#L18-L20).

i.e.

```rb
def index
  generate_codes
  result = BackupCodeSetupForm.new(current_user).submit
  visit_result = result.to_h.merge(analytics_properties_for_visit)
  analytics.backup_code_setup_visit(**visit_result)
  irs_attempts_api_tracker.mfa_enroll_backup_code(success: result.success?)

  save_backup_codes
  track_backup_codes_created
end

def create
  generate_codes
  result = BackupCodeSetupForm.new(current_user).submit
  visit_result = result.to_h.merge(analytics_properties_for_visit)
  analytics.backup_code_setup_visit(**visit_result)
  irs_attempts_api_tracker.mfa_enroll_backup_code(success: result.success?)

  save_backup_codes
  track_backup_codes_created
end
```